ide-support/lsp/lsp-server/src/main/java/io/helidon/lsp/server/management/MavenSupport.java

```diff
@@ -0,0 +1,232 @@
+                        .directory(new File(pomPath).getParentFile())
+                        .verbose(false)
+                        .stdOut(output)
+                        .stdErr(output)
+                        .build()
+                        .execute();
+
+            return CompletableFuture.supplyAsync(() -> {
+                Set<Dependency> result = null;
+                try (
+                        Socket clientSocket = serverSocket.accept();
+                        BufferedReader in = new BufferedReader(new InputStreamReader(clientSocket.getInputStream()));
+                ) {
+                    result = GSON.fromJson(in, new TypeToken<Set<Dependency>>() {
+                    }.getType());
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, "Error when executing the maven command - " + lspMvnDependenciesCommand, e);
+                }
+                LOGGER.log(
+                        Level.FINEST,
+                        "getDependencies() for pom file {0} took {1} seconds",
+                        new Object[]{pomPath, (double) (System.currentTimeMillis() - startTime) / 1000}
+                );
+                return result;
+            }).get(timeout, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            LOGGER.log(Level.SEVERE, "Error when executing the maven command - " + lspMvnDependenciesCommand, e);
+        }
+        LOGGER.log(
+                Level.FINEST,
+                "getDependencies() for pom file {0} took {1} seconds",
+                new Object[]{pomPath, (double) (System.currentTimeMillis() - startTime) / 1000}
+        );
+        return null;
+    }
+
+    private static String getMavenVersion() {
+        final Properties properties = new Properties();
+        final String corePomProperties = "META-INF/maven/io.helidon.ide-support.lsp/io.helidon.lsp.server/pom.properties";
+
+        try (InputStream in = MavenSupport.class.getClassLoader().getResourceAsStream(corePomProperties)) {
+            if (in != null) {
+                properties.load(in);
+            }
+        } catch (IOException ioe) {
+            return "";
+        }
+
+        String version = properties.getProperty("version");
+        return version == null ? "" : version.trim();
+    }
+
+    /**
+     * Get information about all dependencies for the given pom file.
+     *
+     * @param pomPath Path to the pom file.
+     * @return List that contains information about the dependencies.
+     */
+    public Set<Dependency> getDependencies(final String pomPath) {
+        return getDependencies(pomPath, DEFAULT_TIMEOUT);
+    }
+
+    /**
+     * Get pom file for the file from the maven project.
+     *
+     * @param fileName File name.
+     * @return Get pom file for the given file or null if pom.xml is not found.
+     * @throws IOException IOException
+     */
+    public String getPomForFile(final String fileName) throws IOException {
```

Comment:
Can this follow the Helidon style and omit the get prefix. Maybe `resolvePom` is a good name ?